### PR TITLE
Remove vresutils

### LINKED ISSUE #1211: Drop vresutil dependency
## Describe the feature you'd like to see

Drop vresutil dependency and reimplement features.

### PR DESCRIPTION
**Pull request by ekatef**
Closes #1211 replacing `vresutil` dependencies.

To give some context, `vresutil` dependencies have been dropped with #803, but re-appeared due to flaws when merging the sector-coupled version.

## Changes proposed in this Pull Request

- [x] reimplemented annuity calculations
- [x] replaced `vresutil` toolset with shapely ones when calculating cell areas when building the population layouts.

## Checklist

- [x] I consent to the release of this PR's code under the AGPLv3 license and non-code contributions under CC0-1.0 and CC-BY-4.0.
- [x] I tested my contribution locally and it seems to work fine.
- [ ] Code and workflow changes are sufficiently documented.
- [x] Newly introduced dependencies are added to `envs/environment.yaml` and `doc/requirements.txt`.
- [ ] Changes in configuration options are added in all of `config.default.yaml` and `config.tutorial.yaml`.
- [ ] Add a test config or line additions to `test/` (note tests are changing the config.tutorial.yaml)
- [ ] Changes in configuration options are also documented in `doc/configtables/*.csv` and line references are adjusted in `doc/configuration.rst` and `doc/tutorial.rst`.
- [x] A note for the release notes `doc/release_notes.rst` is amended in the format of previous release notes, including reference to the requested PR.

### CODE REVIEW

**Comment by ekatef** [comment]
As a technical comment, the revised implementation for calculating `cell_areas` basically updates the pyproj-based vresutil methods for `pyproj >= 2.1.0`. [There](<https://gis.stackexchange.com/a/127432/118363>) is a nice example on the differences of two approaches. 

Testing has resulted in non-zero differences for the old and new approaches, though the differences are less than 0.5% which is probably fine. To facilitate review, have added a comment block with the old implementation including vresutil part.

**Comment by ekatef** [comment]
@davide-f it would be great to have your review on that as it relates to the fixing the environment issues. Would be great to resolve it before the release

File: scripts/build_population_layouts.py [hunk 1]
```diff
@@ -75,9 +74,26 @@
...
+    # def reproject(geom, fr=pyproj.Proj(proj='longlat'), to=pyproj.Proj(proj='aea', lat_1=33., lat_2=72.)):
+    #     reproject_pts = partial(pyproj.transform, fr, to)
+    #     return transform(reproject_pts, geom)
+
     # in km^2
-    with mp.Pool(processes=snakemake.threads) as pool:
-        cell_areas = pd.Series(pool.map(vshapes.area, grid_cells)) / 1e6
+    # with mp.Pool(processes=snakemake.threads) as pool:
+    #     cell_areas = pd.Series(pool.map(vshapes.area, grid_cells)) / 1e6
+    # ------------------------------------------------------------------------
```
Comment (davide-f) [inline scripts/build_population_layouts.py]: Can we drop this?

**Review by davide-f — COMMENTED** [review]
The PR looks great and CI passes.
It deserves adding a major release probably as the env is changed.
Fine for me then :) great catch!

File: scripts/build_population_layouts.py (hunk 1 above)
Comment (ekatef) [reply scripts/build_population_layouts.py]: I think yes 😄 Done!

**Comment by ekatef** [comment]
> The PR looks great and CI passes. It deserves adding a major release probably as the env is changed. Fine for me then :) great catch!

Thanks a lot @davide-f! The PR just adds a bit to the work you have done to accommodate the fixed environments 🙂 Have added a release note if you think the PR is worth it 😉

Agree, that is time for a major release. If CI will be still happy, I'll merge this PR and start preparations for that, as we discussed. Many thanks for the coordination and all the fantastic work you are doing.

**Pull request merged**